### PR TITLE
🐛Source Auth0 - Fix low code migration bugs

### PR DESCRIPTION
**Pull request by sajarin**
## What
This PR aims to fix bugs introduced in the low code migration of the auth0 source connector

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests#pull-request-title-convention)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

**Comment by sajarin** [comment]
@vasilisgav trying to get this merged as soon as I can

@marcosmarxm can you take a look at the most recent commit when you get a chance, need some advice on how to properly inject the datetimecursor into the query request parameter

File: airbyte-integrations/connectors/source-auth0/source_auth0/manifest.yaml [hunk 1]
```diff
@@ -48,22 +63,48 @@ definitions:
...
+        request_headers: {}
     incremental_sync:
       type: DatetimeBasedCursor
       cursor_field: "updated_at"
       datetime_format: "%Y-%m-%dT%H:%M:%S.%f%z"
       cursor_granularity: "PT0.000001S"
+      start_time_option:
+        inject_into: request_parameter
+        field_name: q
+        type: RequestOption
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-auth0/source_auth0/manifest.yaml:98]: @sajarin are these workign now?

**Review by marcosmarxm — DISMISSED** [review]

File: airbyte-integrations/connectors/source-auth0/source_auth0/manifest.yaml [hunk 2]
```diff
@@ -48,22 +63,48 @@ definitions:
...
+        path: "users"
+        http_method: "GET"
+        authenticator:
+          class_name: source_auth0.components.AuthenticatorAuth0
+          bearer: "#/definitions/bearer_authenticator"
+          oauth: "#/definitions/oauth_authenticator"
+        request_parameters:
+          sort: "updated_at:1"
+          include_totals: "false"
+          # q: "updated_at:[{{stream_interval.get('updated_at') }} TO *]"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-auth0/source_auth0/manifest.yaml:88]: Did you try to use `stream_interval.stream_state`?

**Comment by airbyte-oss-build-runner** [comment]
##  source-auth0 test report (commit [`34936df502`](https://github.com/airbytehq/airbyte/pull/29804/commits/34936df502e73c500e67a9e851172f93edc85810)) - ❌

⏲️  Total pipeline duration: 02mn59s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Connector package install                                           | ✅        |
| Build source-auth0 docker image for platform linux/x86_64           | ✅        |
| Acceptance tests                                                    | ❌        |
| Code format checks                                                  | ❌        |
| Validate airbyte-integrations/connectors/source-auth0/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ✅        |
| QA checks                                                           | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/saj_auth0-fix/1695754189/34936df502e73c500e67a9e851172f93edc85810/source-auth0/0.4.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:34936df502e73c500e67a9e851172f93edc85810)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-auth0 test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-auth0 test report (commit [`3dcb20c587`](https://github.com/airbytehq/airbyte/pull/29804/commits/3dcb20c587c6736256a00889031828dc81e9dabd)) - ❌

⏲️  Total pipeline duration: 01mn13s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Connector package install                                           | ❌        |
| Code format checks                                                  | ✅        |
| Validate airbyte-integrations/connectors/source-auth0/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ✅        |
| QA checks                                                           | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/saj_auth0-fix/1695836017/3dcb20c587c6736256a00889031828dc81e9dabd/source-auth0/0.4.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:3dcb20c587c6736256a00889031828dc81e9dabd)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-auth0 test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-auth0 test report (commit [`b53f452e54`](https://github.com/airbytehq/airbyte/pull/29804/commits/b53f452e54a6609a193fcb051ebf88f2f0e5b44a)) - ❌

⏲️  Total pipeline duration: 01mn07s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Connector package install                                           | ❌        |
| Code format checks                                                  | ✅        |
| Validate airbyte-integrations/connectors/source-auth0/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ✅        |
| QA checks                                                           | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/saj_auth0-fix/1695841265/b53f452e54a6609a193fcb051ebf88f2f0e5b44a/source-auth0/0.4.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:b53f452e54a6609a193fcb051ebf88f2f0e5b44a)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-auth0 test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-auth0 test report (commit [`5b6bb8a5a5`](https://github.com/airbytehq/airbyte/pull/29804/commits/5b6bb8a5a5c93cd547c50cc21215fba2c5545a0e)) - ❌

⏲️  Total pipeline duration: 02mn01s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Connector package install                                           | ✅        |
| Build source-auth0 docker image for platform linux/x86_64           | ✅        |
| Acceptance tests                                                    | ✅        |
| Code format checks                                                  | ❌        |
| Validate airbyte-integrations/connectors/source-auth0/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ✅        |
| QA checks                                                           | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/saj_auth0-fix/1695842080/5b6bb8a5a5c93cd547c50cc21215fba2c5545a0e/source-auth0/0.4.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:5b6bb8a5a5c93cd547c50cc21215fba2c5545a0e)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-auth0 test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-auth0 test report (commit [`5f462011ac`](https://github.com/airbytehq/airbyte/pull/29804/commits/5f462011acfd3a5db0184828ce096423c1cb23d6)) - ✅

⏲️  Total pipeline duration: 01mn57s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Connector package install                                           | ✅        |
| Build source-auth0 docker image for platform linux/x86_64           | ✅        |
| Acceptance tests                                                    | ✅        |
| Code format checks                                                  | ✅        |
| Validate airbyte-integrations/connectors/source-auth0/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ✅        |
| QA checks                                                           | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/saj_auth0-fix/1695842364/5f462011acfd3a5db0184828ce096423c1cb23d6/source-auth0/0.4.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:5f462011acfd3a5db0184828ce096423c1cb23d6)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-auth0 test
```

**Review by archangelic — APPROVED** [review]
LGTM

**Pull request merged**